### ISSUE TITLE
Enable GPU support in MFEM + hypre example.

### ISSUE THREAD

**Issue by balos1** [xsdk-project/xsdk-examples#14]

**Comment by luszczek** [comment]
I get an error in  `mfem/hypre-superlu/CMakeFiles/convdiff.dir/build.make` due to missing variables:
```
mfem/hypre-superlu/convdiff: SUNDIALS::PETSC_VEC-NOTFOUND
mfem/hypre-superlu/convdiff: SUNDIALS::PETSC_SNES-NOTFOUND
```
The message is:
```
mfem/hypre-superlu/CMakeFiles/convdiff.dir/build.make:114: *** target pattern contains no `%'.  Stop.
```
I get the same:
```
mfem/petsc/CMakeFiles/obstacle.dir/build.make:113: *** target pattern contains no `%'.  Stop.
```
and:
```
mfem/sundials/CMakeFiles/transient-heat.dir/build.make:109: *** target pattern contains no `%'.  Stop.
```

**Comment by v-dobrev** [comment]
I added a new GPU-enabled MFEM+HYPRE example in https://github.com/xsdk-project/xsdk-examples/tree/balos1/updates commit 886b8b0.

I think the example `mfem/hypre-superlu/convdiff.cpp` can also be modified slightly to enable GPU support when using HYPRE, at least; usinig GPU-enabled SuperLU in MFEM is not tested and may require some additions in MFEM.

@luszczek, I encountered the same issue as you -- it should be fixed now in this Spack branch: https://gitlab.com/xsdk-project/spack-xsdk/-/tree/xsdk-examples -- it was an issue with a cmake file in SUNDIALS.
